test/runnable/overload.d

```diff
@@ -1226,6 +1226,22 @@ void test14965()
     assert(c.fd2()(113) == 123);
 }
 
+/***************************************************/
+// https://issues.dlang.org/show_bug.cgi?id=21522
+
+struct S21522
+{
+    // alias to template as first symbol in overload set
+    void funA2()(int a) {}
+    void funA2(char a) {}   // function was lost -> OK now
+    alias funA = funA2;
+    void funA(float b) {}
+}
+void test21522()
+{
+    static assert(__traits(getOverloads, S21522, "funA", true).length == 3);
+}
```

Comment:
This looks like it should be a compilable test.

Reply:
There are already a few compile-time tests in this file, the cost of adding it here is 0ms vs creating another.
Unfortunately there's no `test/compilable/overload.d` 😞 .